mne/tests/test_epochs.py

```diff
@@ -1629,19 +1629,41 @@ def test_saved_fname_no_splitting(
     assert not split_1_fpath.is_file()
 
 
+@pytest.mark.parametrize(
+    "epochs_to_split",
+    [
+        ("3MB", 18, False, False, 3),
+        pytest.param(("2GB", 18, False, False, 1), marks=pytest.mark.xfail),
+    ],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "dst_fname",
+    ["test-epo.fif", pytest.param("a_b_c-epo.fif", marks=pytest.mark.xfail)],
+)
+def test_bids_splits_fail_for_bad_fname_ending(epochs_to_split, dst_fname, tmp_path):
+    """Make sure split_naming=bids is only used with bids endings.
+
+    Non-bids endings can cause surprising split names, e.g. test-epo.fif
+    producing splits _split-01_test-epo.fif.
+
+    """
+    epochs, split_size, _ = epochs_to_split
+    dst_fpath = tmp_path / dst_fname
+    save_kwargs = {"split_naming": "bids", "split_size": split_size}
+
+    with pytest.raises(ValueError):
```

Comment:
any `pytest.raises` should have `match=...`, it

1. Makes the test more specific/targeted itself
2. Makes it clearer what the error actually is that you're trying to catch (like a comment)

Reply:
Makes sense, done.